### PR TITLE
Added git-cloning scripts.

### PR DESCRIPTION
**Pull request by harshanL**
This scripts let the Jenkins to enable or disable shallow cloning options.

### CODE REVIEW

File: git-shallow-clone/disable-shallow-cloning.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        CloneOption cloneOption = gitExtensions.get(CloneOption.class)
+        if (cloneOption != null && cloneOption.isShallow()) {
+          gitExtensions.remove(cloneOption)
+          job.save();
+          println("Removed Advanced Cloning options for the job: $job.name")
+        }  
+      } 
+    }
+  } catch (GroovyCastException e) {
+    println("Unable to configure Advanced cloning options for the job:  $job.name")
```
Comment (kasunbg) [inline git-shallow-clone/disable-shallow-cloning.groovy]: Can you log the actual exception as well?

File: git-shallow-clone/disable-shallow-cloning.groovy [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/*Loop through all available jenkins jobs*/
+Jenkins.instance.getItems().each{
+  if (it instanceof Folder) {
+      processFolder(it)
+  } else {
+      disableCloningOptions(it)
+  }
+}
```
Comment (kasunbg) [inline git-shallow-clone/disable-shallow-cloning.groovy]: as a best practice, shall we add a new line at the end ?

File: git-shallow-clone/enable-shallow-cloning.groovy [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      if (gitExtensions != null) {
+        CloneOption cloneOptions = gitExtensions.get(CloneOption.class)
+        if (cloneOptions != null && cloneOptions.isShallow()) {
+          println("Advanced Cloning options has already configured for the job: $job.name")
+        } else {
+          boolean noTags = true
+          boolean shallowClone = true
+          String reference = null
+          int timeOut = 10
+          int depth = 100
```
Comment (kasunbg) [inline git-shallow-clone/enable-shallow-cloning.groovy]: shall we make these top-level constant variables?

File: git-shallow-clone/enable-shallow-cloning.groovy [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          gitExtensions.add(cloneOption)
+          job.save();
+          println("Configured Advanced Cloning options for the job: $job.name")
+        }
+      } else {
+        println("Unable to configure Advanced cloning options for the job:  $job.name")
+      }
+    }
+  } catch (GroovyCastException e) {
+    println("Unable to configure Advanced cloning options for the job:  $job.name")
```
Comment (kasunbg) [inline git-shallow-clone/enable-shallow-cloning.groovy]: log the actual error as well.

File: git-shallow-clone/enable-shallow-cloning.groovy [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        CloneOption cloneOptions = gitExtensions.get(CloneOption.class)
+        if (cloneOptions != null && cloneOptions.isShallow()) {
+          println("Advanced Cloning options has already configured for the job: $job.name")
+        } else {
+          boolean noTags = true
+          boolean shallowClone = true
+          String reference = null
+          int timeOut = 10
+          int depth = 100
+          CloneOption cloneOption =new CloneOption(shallowClone, noTags, reference, timeOut)
```
Comment (kasunbg) [inline git-shallow-clone/enable-shallow-cloning.groovy]: minor formatting issue.

File: git-shallow-clone/disable-shallow-cloning.groovy (hunk 2 above)
Comment (harshanL) [reply git-shallow-clone/disable-shallow-cloning.groovy]: Resolved

File: git-shallow-clone/enable-shallow-cloning.groovy (hunk 3 above)
Comment (harshanL) [reply git-shallow-clone/enable-shallow-cloning.groovy]: Resolved.

**Review by kasunbg — APPROVED** [review]

**Pull request merged**